week-4/E-arrays-of-objects/exercise-3.js

```diff
@@ -59,13 +59,22 @@ var restaurantFinderApplication = {
   applicationVersion: "1.0",
   restaurants: restaurants,
   findAvailableRestaurants: function(numberOfPeople) {
-    // Complete here
+    function availablePlaces(places) {
+      return places.totalSeats - places.numberOfCustomers >= numberOfPeople;
+    }
+    return restaurants.filter(availablePlaces).map(names => names.name);
   },
   findRestaurantServingDish: function(dishName) {
-    // Complete here
+    function dischSurved(menus) {
+      return menus.menu.includes(dishName);
+    }
+    return restaurants.filter(dischSurved).map(names => names.name);
   },
   countNumberOfRestaurantsInArea: function(area) {
-    // Complete here
+    function resturantsInArea(resturant) {
+      return resturant.address.area.includes(area);
+    }
+    return restaurants.filter(resturantsInArea).length;
   }
```

Comment:
Careful with some of the spelling in the code - you'll find many of your future colleagues will tend to be very precise about things being spelled right - but the logic is again very good, good solution.